### PR TITLE
Retire obsolete code in Repo

### PR DESCRIPTION
**Pull request by mih**
- duplicate `get_status()` functionality
- `Repo.is_dirty()` -- hardly used, and replaced by `Repo.status()`
- `Repo.untracked_files` is kept, but reimplemented via `Repo.status()`, but special `AnnexRepo` variant is gone
- tests associated with removed code are gone too

### CODE REVIEW

**Comment by mih** [comment]
I am clueless how these changes affect this, now failing, test: https://travis-ci.org/datalad/datalad/jobs/519927555#L1779

Something in the way that dirty is evaluated, but somehow indirectly. Any hints are most appreciated.

**Comment by kyleam** [comment]
> I am clueless how these changes affect this, now failing, test: https://travis-ci.org/datalad/datalad/jobs/519927555#L1779

It something to do with the addurl batch process state.  The old `ar.get_status()` triggered `ar.precommit()`.  Not a proper solution, but you can see that the test would pass with

```
diff --git a/datalad/support/gitrepo.py b/datalad/support/gitrepo.py
index 810de4938..54f6b1a6f 100644
--- a/datalad/support/gitrepo.py
+++ b/datalad/support/gitrepo.py
@@ -3038,6 +3038,7 @@ def status(self, paths=None, untracked='all', eval_submodule_state='full'):
           `state`
             Can be 'added', 'untracked', 'clean', 'deleted', 'modified'.
         """
+        self.precommit()
         lgr.debug('Query status of %r for %s paths',
                   self, len(paths) if paths else 'all')
         return self.diffstatus(
```

File: datalad/interface/tests/test_download_url.py [hunk 1]
```diff
@@ -126,6 +126,7 @@ def test_download_url_dataset(toppath, topurl, path):
 @with_tempfile(mkdir=True)
 def test_download_url_archive(toppath, topurl, path):
     ds = Dataset(path).rev_create()
+    # TODO opj for URLs?!
```
Comment (kyleam) [inline datalad/interface/tests/test_download_url.py]: Please silently fix up my stupidity rather than bringing attention to it :>

**Comment by mih** [comment]
@kyleam Thx for the pointer! I was blind.

**Comment by mih** [comment]
I added the precommit to `GitRepo.get_content_info(ref=None)` to catch any query on the work tree state. I hope this catches all relevant cases and, at the same time, minimizes performance impact. Entirely based on gut feeling.

**Comment by mih** [comment]
> Looks good, especially this part:
> 
> ```
> 34 insertions(+), 448 deletions(-)
> ```

I have high hopes for #3333 too!

**Pull request merged**